pkg/plugin/driver/daemonset/daemonset.go

```diff
@@ -158,6 +189,30 @@ func (p *Plugin) listOptions() metav1.ListOptions {
 	}
 }
 
+func (p *Plugin) getSecretName() string {
+	return fmt.Sprintf("daemonset-%s-%s", p.GetName(), p.SessionID)
+}
+
+func (p *Plugin) createClientSecret(client kubernetes.Interface, key crypto.PrivateKey) error {
+	rsaKey, ok := key.(*rsa.PrivateKey)
+	if !ok {
+		return errors.New("private key not RSA")
+	}
+
+	bytes := x509.MarshalPKCS1PrivateKey(rsaKey)
+	secretName := p.getSecretName()
+
+	secret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: p.Namespace,
+		},
+		Data: map[string][]byte{clientKeyName: bytes},
```

Comment:
I don't really know if there's a compelling reason to use it, but there's a standard Kubernetes Secret `type` for TLS keypairs:
```yaml
apiVersion: v1
kind: Secret
metadata:
  creationTimestamp: null
  name: server-cert
  namespace: heptio-raven
type: kubernetes.io/tls
data:
  tls.crt: "<certificate + chain in PEM format>"
  tls.key: "<private key in PEM format>"
```

I think may be valuable to put the certificate alongside the private key in a Secret. Even though the certificate itself isn't secret, it's tightly bound to the private key so you almost always want them to change as an atomic unit.